### PR TITLE
Feature: Add telemetry for the Event block

### PR DESCRIPTION
**Pull request by kjohnson**
Resolves [GIVE-505](https://stellarwp.atlassian.net/browse/GIVE-505).
Resolves [GIVE-506](https://stellarwp.atlassian.net/browse/GIVE-506).

Related https://github.com/impress-org/give-telemetry-server/pull/18.

## Description

This PR introduces 2 new data points to the telemetry (tracking) system:

- `DonationFormsData` -> `event_block`
- `GivePluginSettingsData` -> `is_event_tickets`

Supporting this change is a new `Helpers/FormBlocks` class, which abstracts access to the `BlockCollection` to confirm a block's existence by name.

[GIVE-505]: https://stellarwp.atlassian.net/browse/GIVE-505?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[GIVE-506]: https://stellarwp.atlassian.net/browse/GIVE-506?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Tracking/TrackingData/DonationFormsData.php [hunk 1]
```diff
@@ -81,6 +83,7 @@ protected function getData()
                 'form_template' => $this->getFormTemplate($formId),
                 'donor_count' => $this->formDonorCounts[$formId],
                 'revenue' => $this->formRevenues[$formId],
+                'event_block' => FormBlocks::formId($formId)->hasBlock('give_event_block'),
```
Comment (pauloiankoski) [inline src/Tracking/TrackingData/DonationFormsData.php:86]: Would it be `givewp/event-tickets` here?

Comment (kjohnson) [reply src/Tracking/TrackingData/DonationFormsData.php:86]: Ah, copy/paste error. I'll fix that.

Comment (kjohnson) [inline src/Tracking/TrackingData/DonationFormsData.php:86]:
```suggestion
                'event_block' => FormBlocks::formId($formId)->hasBlock('givewp/event-tickets'),
```

Comment (kjohnson) [reply src/Tracking/TrackingData/DonationFormsData.php]: Resolved, see https://github.com/impress-org/givewp/pull/7341/commits/96706297af9d132094a7b8ea287a08886365b026.

**Review by pauloiankoski — APPROVED** [review]
LGTM!

**Comment by kjohnson** [comment]
Build for QA, see https://github.com/impress-org/givewp/actions/runs/8742421542.

**Comment by kjohnson** [comment]
Build for QA, see https://github.com/impress-org/givewp/actions/runs/8836423379

File: src/Tracking/TrackingData/DonationFormsData.php [hunk 2]
```diff
@@ -81,6 +83,7 @@ protected function getData()
                 'form_template' => $this->getFormTemplate($formId),
                 'donor_count' => $this->formDonorCounts[$formId],
                 'revenue' => $this->formRevenues[$formId],
+                'event_block' => FormBlocks::formId($formId)->hasBlock('givewp/event-tickets'),
```
Comment (kjohnson) [inline src/Tracking/TrackingData/DonationFormsData.php:86]: After some troubleshooting on a live site, it seems that this `bool` needs to be manually cast to an `int` (`1` or `0`) before being sent to the server (which the server will then re-cast as a `bool`).

Comment (kjohnson) [reply src/Tracking/TrackingData/DonationFormsData.php:86]:
```suggestion
                'event_block' => (int) FormBlocks::formId($formId)->hasBlock('givewp/event-tickets'),
```

**Comment by kjohnson** [comment]
Build for QA, see https://github.com/impress-org/givewp/actions/runs/8838257778

**Pull request merged**